### PR TITLE
SW-4749 /SW-4751 Add project to planting site create/edit/view refactored component

### PR DESCRIPTION
**Pull request by karthikbtf**
- added project to planting site, moved useProjects and ProjectsDropdown to root level directories
- extracted out all the planting site input into a separate component (this effort started out as a refactor and repurposed it to add the project as well)
- the extracted out component will be reused in user detailed sites
- bumped web -components

Terraware App 2024-01-17 11-05-59

Terraware App 2024-01-17 10-59-02

Terraware App 2024-01-17 10-58-46

### CODE REVIEW

File: src/components/PlantingSites/DetailsInputForm.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            showSaveValidationErrors={showSaveValidationErrors}
+            timeZone={effectiveTimeZone}
+          />
+        </Grid>
+      )}
+      {projectsEnabled && (
+        <Grid item xs={gridSize()}>
+          <ProjectsDropdown<PlantingSite> availableProjects={availableProjects} record={record} setRecord={setRecord} />
+        </Grid>
+      )}
```
Comment (karthikbtf) [inline src/components/PlantingSites/DetailsInputForm.tsx:132]: this is new code

File: src/components/PlantingSites/DetailsInputForm.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  const gridSize = () => {
+    if (isMobile) {
+      return 12;
+    }
+    return 4;
+  };
+
+  return (
+    <Grid container display='flex' spacing={3} flexGrow={0}>
```
Comment (karthikbtf) [inline src/components/PlantingSites/DetailsInputForm.tsx:82]: this was extracted out from PlantingSiteCreate

File: src/components/PlantingSites/PlantingSiteCreate.tsx [hunk 3]
```diff
@@ -194,53 +181,14 @@ export default function CreatePlantingSite(props: CreatePlantingSiteProps): JSX.
                 </Box>
                 <PageSnackbar />
                 <Card flushMobile style={{ flexGrow: 1, display: 'flex', flexDirection: 'column' }}>
-                  <Grid container display='flex' spacing={3} flexGrow={0}>
```
Comment (karthikbtf) [inline src/components/PlantingSites/PlantingSiteCreate.tsx:197]: moved to `DetailsInputForm`

File: src/components/InventoryV2/form/BatchDetailsForm.tsx [hunk 4]
```diff
@@ -16,6 +16,8 @@ import { stateName } from 'src/types/Accession';
 import isEnabled from 'src/features';
 import { SavableBatch } from 'src/redux/features/batches/batchesAsyncThunks';
 import useSnackbar from 'src/utils/useSnackbar';
+import { useProjects } from 'src/hooks/useProjects';
```
Comment (karthikbtf) [inline src/components/InventoryV2/form/BatchDetailsForm.tsx:19]: Moved `useProjects` into `src/hooks`.. if this is better off residing alongside `ProjectsDropdown` I can make that change. Picked `src/hooks` as a starting point for now.

**Review by nickgraz — APPROVED** [review]
Nice work, thanks for moving that hook to a better spot!

**Pull request merged**